Doppler.AccountPlans/Utils/CalculateUpgradeCostHelper.cs

```diff
@@ -7,7 +7,7 @@ public static class CalculateUpgradeCostHelper
     {
         public static PlanAmountDetails CalculatePlanAmountDetails(PlanInformation newPlan, PlanDiscountInformation newDiscount, PlanInformation currentPlan, DateTime now)
         {
-            var currentBaseMonth = now.Day >= 21 ? currentPlan.CurrentMonthPlan : currentPlan.CurrentMonthPlan - 1;
+            var currentBaseMonth = now.Day >= 21 ? currentPlan.CurrentMonthPlan != 0 ? currentPlan.CurrentMonthPlan : 1 : currentPlan.CurrentMonthPlan - 1;
```

Comment:
Cuando anidas ternaries, está bueno usar parentesis o varias líneas para que se pueda entender mejor sin perderse. 

Y estoy viendo que también conviene pasar las condiciones para el siguiente término, algo así:

```
var resultado = condición_1 ? valor_para_condición_1
    : condición_2 ? valor_para_condición_2
    : condición_3 ? valor_para_condición_3
    : valor_para_cuando_no_se_dan_condiciones_1_ni_2_ni_3;
```

Con lo que escribiste me parece que lo mejor es reordenar las condiciones o usar `&&` o `||` en lugar de concatenar condiciones con el ternario, a ver si me sale (ojo, no estoy seguro de entender la condición original, por ahí lo interpreto mal)

```suggestion
            var currentBaseMonth = now.Day >= 21 && currentPlan.CurrentMonthPlan != 0 ? currentPlan.CurrentMonthPlan
                : currentPlan.CurrentMonthPlan == 0 ? 1 
                : currentPlan.CurrentMonthPlan - 1;
```

```suggestion
            var currentBaseMonth = currentPlan.CurrentMonthPlan == 0 ? 1 
                : now.Day >= 21 ? currentPlan.CurrentMonthPlan
                : currentPlan.CurrentMonthPlan - 1;
```